src/datasource.ts

```diff
@@ -80,12 +83,13 @@ export default class VividCortexDatasource {
    * @param  {object} target
    * @param  {number} from
    * @param  {number} until
+   * @param  {number} maxDataPoints
    * @return {Promise}
    */
-  doQuery(target: any, from: number, until: number) {
+  doQuery(target: any, from: number, until: number, maxDataPoints: number) {
     const params = {
       from: from,
-      samplesize: 12,
+      samplesize: Math.max(Math.floor((until - from) / maxDataPoints), 1),
```

Comment:
can we cover this with a test ?

Reply:
Absolutely.

Reply:
Done. Let me know if you think of other scenarios to test.